internal/api/cluster.go

```diff
@@ -130,6 +131,49 @@ func handleRetryCreateCluster(c *Context, w http.ResponseWriter, r *http.Request
 	outputJSON(c, w, cluster)
 }
 
+// handleProvisionCluster responds to POST /api/cluster/{cluster}/provision,
+// provisioning k8s resources on a previously-created cluster.
+func handleProvisionCluster(c *Context, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	clusterID := vars["cluster"]
+	c.Logger = c.Logger.WithField("cluster", clusterID)
+
+	cluster, status, unlockOnce := lockCluster(c, clusterID)
+	if status != 0 {
+		w.WriteHeader(status)
+		return
+	}
+	defer unlockOnce()
+
+	switch cluster.State {
+	case model.ClusterStateStable:
+	case model.ClusterStateProvisioningFailed:
+	case model.ClusterStateProvisioningRequested:
+	default:
+		c.Logger.Warnf("unable to provision cluster while in state %s", cluster.State)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if cluster.State != model.ClusterStateProvisioningRequested {
```

Comment:
We already checked this above.

Reply:
Yeah, although this check feels a bit redundant, the purpose of it is to only update the provisioner database if the state is anything other than `ClusterStateProvisioningRequested`. It saves some code space by not duplicating the code under each case statement.